common/src/main/java/com/viaversion/viabackwards/api/rewriters/LegacyEnchantmentRewriter.java

```diff
@@ -95,22 +102,21 @@ public void rewriteEnchantmentsToClient(CompoundTag tag, boolean storedEnchant)
         }
         if (!lore.isEmpty()) {
             if (!storedEnchant && enchantments.isEmpty()) {
-                CompoundTag dummyEnchantment = new CompoundTag();
-                dummyEnchantment.putShort("id", (short) 0);
-                dummyEnchantment.putShort("lvl", (short) 0);
-                enchantments.add(dummyEnchantment);
-
-                tag.put(nbtTagName + "|dummyEnchant", new ByteTag());
-
-                NumberTag hideFlags = tag.getNumberTag("HideFlags");
-                if (hideFlags == null) {
-                    hideFlags = new IntTag();
-                } else {
-                    tag.putInt(nbtTagName + "|oldHideFlags", hideFlags.asByte());
+                CompoundTag dummyEnchantment = getDummyEnchantment();
+                if (dummyEnchantment != null) {
```

Comment:
why the null check?

Reply:
So that the platform can remove the adding of the dummy enchantment

Reply:
Make a protected boolean field for that instead then